imports/validators.py

```diff
@@ -23,22 +29,61 @@ def __call__(self, value):
             )
 
         with zipfile.ZipFile(value.file) as zfile:
-            required_extensions = ('.shp', '.shx', '.dbf')
-            filenames, extensions = [], []
-            for filename in zfile.namelist():
-                if filename.endswith(required_extensions):
-                    filename, extension = os.path.splitext(filename)
-                    filenames.append(filename)
-                    extensions.append(extension)
-
-            if len(set(filenames)) > 1:
-                raise ValidationError(
-                    self.messages['mismatched_filenames'],
-                    code='mismatched_filenames',
-                )
-
-            if set(extensions) != set(required_extensions):
-                raise ValidationError(
-                    self.messages['missing_required_files'],
-                    code='missing_required_files',
-                )
+            self._validate_files(zfile)
+
+            dbf, shp, shx = sorted([name for name in zfile.namelist() if name.endswith(REQUIRED_SHAPEFILE_EXTENSIONS)])
+            with zfile.open(shp) as shp, zfile.open(shx) as shx, zfile.open(dbf) as dbf:
+                shapefile_reader = shapefile.Reader(shp=shp, shx=shx, dbf=dbf)
+
+            self._validate_fields(shapefile_reader)
+            self._validate_feature_classes(shapefile_reader)
+
+    def _validate_files(self, zfile):
+        filenames, extensions = [], []
+        for filename in zfile.namelist():
+            if filename.endswith(REQUIRED_SHAPEFILE_EXTENSIONS):
+                filename, extension = os.path.splitext(filename)
+                filenames.append(filename)
+                extensions.append(extension)
+
+        if len(set(filenames)) > 1:
+            raise ValidationError(
+                self.messages['mismatched_filenames'],
+                code='mismatched_filenames',
+            )
+
+        if set(extensions) != set(REQUIRED_SHAPEFILE_EXTENSIONS):
+            raise ValidationError(
+                self.messages['missing_required_files'],
+                code='missing_required_files',
+            )
+
+    def _validate_fields(self, shapefile_reader):
+        shapefile_fields = [field[0] for field in shapefile_reader.fields[1:]]
```

Comment:
This needs comments. Why are you not taking the first field shapefile_reader.fields? Why are you only taking the first entry from fields? I have no idea what is stored in the `shapefile_fields` here :)

Reply:
This is due to the format of `.dbf` files that reserves the first column is a `DeletionFlag`, but you are right there, needs to add a comment here.